devserver/concatjs/concatjs.go

```diff
@@ -96,25 +96,30 @@ func acceptGzip(h http.Header) bool {
 // FileSystem is the interface to reading files from disk.
 // It's abstracted into an interface to allow tests to replace it.
 type FileSystem interface {
-	statMtime(filename string) (time.Time, error)
-	readFile(filename string) ([]byte, error)
+	StatMtime(filename string) (time.Time, error)
+	ReadFile(filename string) ([]byte, error)
+	ResolvePath(root string, file string) (string, error)
 }
 
 // realFileSystem implements FileSystem by actual disk access.
-type realFileSystem struct{}
+type RealFileSystem struct {}
 
-func (fs *realFileSystem) statMtime(filename string) (time.Time, error) {
+func (fs *RealFileSystem) StatMtime(filename string) (time.Time, error) {
 	s, err := os.Stat(filename)
 	if err != nil {
 		return time.Time{}, err
 	}
 	return s.ModTime(), nil
 }
 
-func (fs *realFileSystem) readFile(filename string) ([]byte, error) {
+func (fs *RealFileSystem) ReadFile(filename string) ([]byte, error) {
 	return ioutil.ReadFile(filename)
 }
 
+func (fs *RealFileSystem) ResolvePath(root string, file string) (string, error) {
+	return filepath.Join(root, file), nil;
```

Comment:
What do you mean by `the user can just do it themselves`? We need some layer of abstraction so that we can specify a custom resolving logic:

* For Bazel, where paths are resolved through the runfiles
* For default (backwards compatibility), where paths are just joined (like here)